src/app/server/Server.cpp

```diff
@@ -302,8 +290,10 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
     }
 #endif // CHIP_CONFIG_ENABLE_SERVER_IM_EVENT
 
-    // This initializes clusters, so should come after lower level initialization.
-    InitDataModelHandler();
+    // Set the data model provider, initializing it in the process. This must
+    // happen after SetSafeAttributePersistenceProvider to ensure the provider
+    // can utilize the safe persistence provider for its attribute storage.
```

Comment:
It must happen after a lot more than that, which was the whole discussion in this PR.  It needs to happen after all low-level bits cluster implementations might use as they are initialized have been initialized.

In particular, we need to make it clear in the comments that this callsite is where cluster implementations will be initialized.  The previous comment, which this PR is removing, did that.

Reply:
Moved this up after the fabric table is set

Reply:
That's not what the comment was about.